### PR TITLE
 [#162537825] Search items

### PR DESCRIPTION
**Pull request by tri-dang**
- Redesign action buttons: `search`, `home`, `create location`
- Add search bar which `back` button and `clear` button
- Add search results page
- Use full text search of `postgres` for search on `name` and `description` of item
- Add `geokit-rails` to support query by distance
- Show distance in search result
- Use cookie to store user's location
- When going to search page, it will return nearest items even not type any text, this will helpful in future when we want to promote any items
- Add `kaminari` to paginator
- Scroll down to load more

Screen Shot 2019-06-22 at 22 25 19
Screen Shot 2019-06-22 at 22 36 25

### CODE REVIEW

File: app/javascript/map.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ class Map {
       },
```
Comment (tri-dang) [inline app/javascript/map.js]: prettier did change, not me :(

File: app/javascript/not-jquery.js [hunk 2]
```diff
@@ -2,16 +2,25 @@
...
+    if (document.readyState != 'loading') {
       fn();
     } else {
       document.addEventListener('DOMContentLoaded', fn);
     }
-  }
-
-}
+  },
+  callAjax: function(url, callback) {
```
Comment (tri-dang) [inline app/javascript/not-jquery.js]: it seems that we do not prefer `jQuery` so I tried to write in pure js.

Comment (tri-dang) [reply app/javascript/not-jquery.js]: not working with Safari :(

File: app/views/search/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      isCalling = false;
+    }
+  }
+
+  function listen() {
+    height = document.documentElement.offsetHeight,
+		scrollingElement = document.scrollingElement || document.documentElement;
+    offset = scrollingElement.scrollTop + window.innerHeight;
+
+    if (!isCalling && offset >= height - 200) {
```
Comment (tri-dang) [inline app/views/search/index.html.erb]: stupid way to do load more, if there is any better js lib you wanna use, I am happy to change.

**Comment by sillylogger** [comment]
Awesome!  I renamed some stuff, simplified the CSS a bit... hopefully I didn't overlook something.

**Comment by sillylogger** [comment]
My notes when reviewing the code were:

##### app/views/layouts/application.html.erb
- font-awesome.. necessary? just for the icon?  we already have other icons :-/
- ah, I see you removed it already :-)
- href="/search?back_path", let's use rails routes instead  
  
##### app/views/search/index.html.erb
- class="items__item__content__price" woahhh.. I try to use just `block-name__element-name m--modifier`
- listen & callback are bad names :-P
  
##### app/views/search/_form.html.erb
- form uses action="/search", let's make this a route instead

**Pull request merged**